msgraph/core/middleware/redirect.py

```diff
@@ -0,0 +1,38 @@
+import typing
+from http import client
+
+import httpx
+from kiota_http.middleware import BaseMiddleware, RedirectHandler
+
+from .._enums import FeatureUsageFlag
+from .middleware import GraphRequest
+
+
+class GraphRedirectHandler(RedirectHandler):
```

Comment:
do we still need this at all since we should already have it in kiota http?

Reply:
Extends the one in `kiota_http` to add graph specific stuff such as `request context` and setting `FeatureUsageFlags`

Reply:
Just for my benefit, is there any other extra customization being done apart from `FeatureUsageFlags` and `request context`.
I believe that `FeatureUsageFlags` can all be inferred at one place like the telemetry handler by simply checking the items in the middleware collection or computed once beforehand when the client is initialized.
Similarly, `request context` can be copied over to new requests if its held within a [`RequestOption`](https://github.com/microsoft/kiota-abstractions-python/blob/main/kiota_abstractions/request_option.py) instance.
This way we can have only one redirect handler from the core abstractions.

Reply:
thanks for the additional context, please do so Python implementation is not diverging from other languages.